File: ANUL 2/MEDII DE PROIECTARE SI PROGRAMARE/LaboratoareMPP/FestivalDeMuzicaCSharpVarianta2/Validation/TicketValidator.cs
using System.Security.Cryptography;
using FestivalDeMuzicaCSharp.Domain;

namespace FestivalDeMuzicaCSharp.Validation;

public class TicketValidator : IValidator<Ticket>
{
    public void Validate(Ticket entity)
    {
        string errors = "";

        if (string.IsNullOrEmpty(entity.ClientName))
            errors += "Invalid client name!\n";
        
        if (!string.IsNullOrEmpty(errors))
            throw new Exception(errors);
    }
}

File: ANUL 2/METODE AVANSATE DE PROGRAMARE/Zboruri/src/main/java/com/example/zboruri/Validator/TicketValidator.java
package com.example.zboruri.Validator;

import com.example.zboruri.Domain.Ticket;

public class TicketValidator implements AbstractValidator<Ticket> {
    @Override
    public void validate(Ticket ticket) {
        String errors = "";
        if (!errors.isEmpty())
            throw new RuntimeException(errors);
    }
}
